pkg/iosrc/uri.go

```diff
@@ -0,0 +1,87 @@
+package iosrc
+
+import (
+	"errors"
+	"net/url"
+)
+
+type URI struct {
+	url.URL
+}
```

Comment:
Maybe just `type URI url.URL`?

Reply:
My gut is tell me that I wrote it this way for a particular reason, but now that I'm looking at it I can't see why. I'll do as you suggest.